src/Libraries/RevitNodes/GeometryConversion/ProtoToRevitMesh.cs

```diff
@@ -102,35 +101,42 @@ public static class ProtoToRevitMesh
              ElementId MaterialId = null,
             bool performHostUnitConversion = true)
         {
-         
+
             var verts = mesh.VertexPositions;
-            var indicies = mesh.FaceIndices;
+            var indices = mesh.FaceIndices.ToList();
 
             var currentVerts = new List<Autodesk.DesignScript.Geometry.Point>();
             var tsb = new TessellatedShapeBuilder();
             tsb.OpenConnectedFaceSet(false);
 
-            foreach (var f in indicies)
+            for (int faceindex = 0; faceindex < indices.Count(); faceindex++)
             {
-
+                var f = indices[faceindex];
                 currentVerts.Clear();
-                var currentIndicies = new List<uint>() { f.A, f.B, f.C, f.D };
-                for (int i = 0; i < f.Count; i++)
+                var currentIndices = new List<uint>() { f.A, f.B, f.C, f.D };
+
+                //if this is a quad face triangulate it and skip the rest of this loop
+                if (f.Count > 3)
                 {
-                    var currentindex = Convert.ToInt32(currentIndicies[i]);
+                    indices.Add(IndexGroup.ByIndices(f.B, f.C, f.A));
+                    indices.Add(IndexGroup.ByIndices(f.A, f.C, f.D));
+                    continue;
```

Comment:
Hate to be the performance guy, but is there a strong reason to add to the original list while iterating over it? Your iteration will terminate, but the code is a bit strange to read and memory intensive. Consider the case where you're dealing with a 1 million face quad mesh - now you have 3 millions index groups in this list.